src/ast/nodes/nodes_loadValueInRegister.ipp

```diff
@@ -0,0 +1,26 @@
+#include "nodes.h"
+
+namespace ast {
+
+inline void BinaryOperation::loadValueInRegister(codegen::NasmGenerator_x86_64 &generator, scopes::GeneralPurposeRegister targetRegister) const {
+  auto gRhsRegister = generator.regSet().acquireGuard();
```

Comment:
just a question here, is it an application of the RAII concept?

Reply:
yep